test/extensions/quic_listeners/quiche/integration/quic_http_integration_test.cc

```diff
@@ -0,0 +1,186 @@
+    auto connection = std::make_unique<EnvoyQuicClientConnection>(
+        getNextServerDesignatedConnectionId(), server_addr, conn_helper_, alarm_factory_,
+        quic::ParsedQuicVersionVector{supported_versions_[0]}, local_addr, *dispatcher_, nullptr);
+    auto session = std::make_unique<EnvoyQuicClientSession>(
+        quic_config_, supported_versions_, std::move(connection), server_id_, &crypto_config_,
+        &push_promise_index_, *dispatcher_, 0);
+    session->Initialize();
+    return session;
+  }
+
+  // This call may fail because of INVALID_VERSION, because QUIC connection doesn't support
+  // in-connection version negotiation.
+  // TODO(#8479) Propagate INVALID_VERSION error to caller and let caller to use server advertised
+  // version list to create a new connection with mutually supported version and make client codec
+  // again.
+  IntegrationCodecClientPtr makeRawHttpConnection(Network::ClientConnectionPtr&& conn) override {
+    IntegrationCodecClientPtr codec = HttpIntegrationTest::makeRawHttpConnection(std::move(conn));
+    ASSERT(!codec->connected());
+    dynamic_cast<EnvoyQuicClientSession*>(codec->connection())->cryptoConnect();
+    if (codec->disconnected()) {
+      // Connection may get closed during version negotiation or handshake.
+      ENVOY_LOG(error, "Fail to connect to server with error: {}",
+                codec->connection()->transportFailureReason());
+    } else {
+      codec->setCodecClientCallbacks(client_codec_callback_);
+    }
+    return codec;
+  }
+
+  quic::QuicConnectionId getNextServerDesignatedConnectionId() {
+    quic::QuicCryptoClientConfig::CachedState* cached = crypto_config_.LookupOrCreate(server_id_);
+    // If the cached state indicates that we should use a server-designated
+    // connection ID, then return that connection ID.
+    quic::QuicConnectionId conn_id = cached->has_server_designated_connection_id()
+                                         ? cached->GetNextServerDesignatedConnectionId()
+                                         : quic::EmptyQuicConnectionId();
+    return conn_id.IsEmpty() ? quic::QuicUtils::CreateRandomConnectionId() : conn_id;
+  }
+
+  void initialize() override {
+    config_helper_.addConfigModifier([](envoy::config::bootstrap::v2::Bootstrap& bootstrap) {
+      envoy::api::v2::auth::DownstreamTlsContext tls_context;
+      ConfigHelper::initializeTls({}, *tls_context.mutable_common_tls_context());
+      auto* filter_chain =
+          bootstrap.mutable_static_resources()->mutable_listeners(0)->mutable_filter_chains(0);
+      auto* transport_socket = filter_chain->mutable_transport_socket();
+      TestUtility::jsonConvert(tls_context, *transport_socket->mutable_config());
+    });
+    config_helper_.addConfigModifier(
+        [](envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager&
+               hcm) {
+          hcm.mutable_delayed_close_timeout()->set_nanos(0);
```

Comment:
can this go in QUIC_HTTP_PROXY_CONFIG?

Reply:
Yes. But this config is to avoid a NOT_IMPLEMENTED crash because QUIC doesn't implement delayed close yet. I would prefer not to put it in QUIC_HTTP_PROXY_CONFIG which people may regard as a sample config, because this setting is just a temporary thing and eventually we want to support this.